recipes-rubygems/rubygems-aws-sdk-cloudwatchlogs_1.44.0.bb

```diff
@@ -0,0 +1,195 @@
+        "OBJDUMP": d.expand("${OBJDUMP}"),
+        "RANLIB": d.expand("${RANLIB}"),
+        "STRIP": d.expand("${STRIP}"),
+        "cppflags": d.expand("${CPPFLAGS}"),
+        "SOEXT": d.expand("so.${PV}"),
+        "DLEXT": d.expand("so.${PV}"),
+    }
+
+    with open(d.expand("${RUBYLIB}/rbconfig.rb")) as i:
+        cnt = i.read()
+
+    for m in re.finditer(r'^(\s+|\t+)CONFIG\[\"(?P<var>.*)\"\]\s+=\s+\"(?P<value>.*)\"$', cnt, re.MULTILINE):
+        if m.group("var") in _map:
+            _rpl = '  CONFIG["{}"] = "{}"'.format(m.group("var"), _map[m.group("var")])
+            bb.note("Replace {} by {}".format(m.group(0), _rpl))
+            cnt = cnt.replace(m.group(0), _rpl)
+
+    with open(d.expand("${RUBYLIB}/rbconfig.rb"), "w") as o:
+        o.write(cnt)
+}
+
+python do_rubygems_fix_libs() {
+    # as ruby dynloader expects the shared .so files
+    # without extension we will create symlinks to the
+    # properly versioned file
+    for root, dirs, files in os.walk(d.expand("${GEM_HOME}")):
+        for f in files:
+            if f.endswith(d.expand("so.${PV}")):
+                _filename = os.path.basename(f)
+                while "." in _filename:
+                    try:
+                        os.symlink(os.path.basename(f), os.path.join(root, _filename))
+                    except:
+                        pass
+                    _filename, _ = os.path.splitext(_filename)
+}
+
+python do_rubygems_gen_test() {
+    import glob
+    import json
+
+    def sanitize_name(pn):
+        return pn.replace("-", "_")
+
+    _tpl_class = rubygem_load_template(d, "rubygems_testclass.template")
+    _tpl_require = rubygem_load_template(d, "rubygems_testrequire.template")
+    _tpl_gem = rubygem_load_template(d, "rubygems_testgemlist.template")
+    _tpl_exec = rubygem_load_template(d, "rubygems_execwrapper.template")
+
+    __tests = set()
+
+    for _file in glob.glob(d.expand("${PKGDEST}/${PN}/${libdir}/ruby/gems/${GEMLIB_VERSION}/gems/*/lib/*.rb")):
+        _filename, _ext = os.path.splitext(_file)
+        _filename = os.path.basename(_filename)
+        __tests.add(_tpl_require.format(
+            pn=sanitize_name(d.getVar("BPN")),
+            require=_filename,
+            sanitize_name=sanitize_name(_filename).replace(".", "_")))
+
+    for _file in glob.glob(d.expand("${PKGDEST}/${PN}/${bindir}/*")):
+        _filename = os.path.basename(_file)
+        __tests.add(_tpl_exec.format(
+            sanitize_app=sanitize_name(_filename),
+            exec=_filename))
+
+    __tests.add(_tpl_gem.format(
+        pn=sanitize_name(d.getVar("BPN")),
+        gemname=d.getVar("GEM_NAME")))
+
+    with open(d.expand("${RUBYGEMS_LAYERDIR}/lib/oeqa/runtime/cases/rubygems_%s.py" % sanitize_name(d.getVar("BPN"))), "w") as o:
+        o.write(_tpl_class.format(
+                    pn=sanitize_name(d.getVar("BPN")),
+                    tests="\n\n".join(sorted(__tests))))
+        o.write("\n")
+}
+
+python() {
```

Comment:
potential issue: Multiple fragments of the same function in the same file should be merged